src/lava/lib/optimization/problems/problems.py

```diff
@@ -187,90 +196,89 @@ class QP:
 
     def __init__(
         self,
-        hessian: np.ndarray,
+        hessian: npt.ArrayLike,
         linear_offset: ty.Optional[np.ndarray] = None,
         constraint_hyperplanes: ty.Optional[np.ndarray] = None,
         constraint_biases: ty.Optional[np.ndarray] = None,
         constraint_hyperplanes_eq: ty.Optional[np.ndarray] = None,
         constraint_biases_eq: ty.Optional[np.ndarray] = None,
     ):
-        if (
-            constraint_hyperplanes is None and constraint_biases is not None
-        ) or (
-            constraint_hyperplanes is not None and constraint_biases is None
-        ):

```

Comment:
Working, but way prettier to use Pythons XOR operator:
if (constraint_hyperplanes is None) ^ (constraint_biases is None)